### PR TITLE
A few cosmetic improvements to code & comments in liballoc and libcore

### PR DESCRIPTION
**Pull request by alexreg**
Factored out from hacking on rustc for work on the REPL.

r? @Centril

### CODE REVIEW

**Comment by JohnCSimon** [comment]
Ping from triage:

@Centril Can you please review this?

CC @alexreg 


Thanks!

**Comment by alexreg** [comment]
@Centril Yeah, these are a pretty small number of changes mainly focused on doc comment conventions, so I think we're safe merging this, once you've checked it over. :-)

File: src/liballoc/raw_vec.rs [hunk 1]
```diff
@@ -167,13 +167,13 @@ impl<T, A: Alloc> RawVec<T, A> {
...
-    /// Reconstitutes a RawVec from a pointer, capacity.
+    /// Reconstitutes a `RawVec` from a pointer and capacity.
     ///
     /// # Undefined Behavior
     ///
-    /// The ptr must be allocated (on the system heap), and with the given capacity. The
-    /// capacity cannot exceed `isize::MAX` (only a concern on 32-bit systems).
-    /// If the ptr and capacity come from a RawVec, then this is guaranteed.
+    /// The `ptr` must be allocated (on the system heap), and with the given `capacity`.
+    /// Th `capacity` cannot exceed `isize::MAX` (only a concern on 32-bit systems).
```
Comment (Centril) [inline src/liballoc/raw_vec.rs]:
```suggestion
    /// The `capacity` cannot exceed `isize::MAX` (only a concern on 32-bit systems).
```

**Review by Centril — COMMENTED** [review]
Since this also contains improvements to the public standard library docs I think we can land this and it is more "reasonably sized".

**Comment by Centril** [comment]
(r=me with typo ^--- fixed)

**Comment by alexreg** [comment]
Thanks.

@bors r=centril

**Pull request merged**